catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/GeoPdfParser.java

```diff
@@ -0,0 +1,292 @@
+                                "No projection array found on the map frame.  Map Frame will be skipped.");
+                    }
+                }
+            // Handle One Map Frame
+            } else if (lgiDictObject instanceof COSDictionary) {
+                COSDictionary lgidict = (COSDictionary) lgiDictObject;
+                COSDictionary projectionArray = (COSDictionary) lgidict.getDictionaryObject(
+                        PROJECTION);
+                if (projectionArray != null) {
+                    String projectionType =
+                            ((COSString) projectionArray.getItem(PROJECTION_TYPE)).getString();
+                    if (GEOGRAPHIC.equals(projectionType)) {
+                        COSArray neatlineArray = (COSArray) pdPage.getCOSObject()
+                                .getObjectFromPath(LGIDICT + "/" + NEATLINE);
+                        if (neatlineArray == null) {
+                            neatlineArray = generateNeatLineFromPDFDimensions(pdPage);
+
+                        }
+                        polygons.add(getWktFromNeatLine(lgidict, neatlineArray, toDoubleVisitor));
+
+                    } else {
+                        LOGGER.debug("Unsupported projection type {}.  Map Frame will be skipped.",
+                                projectionType);
+                    }
+                } else {
+                    LOGGER.debug(
+                            "No projection array found on the map frame.  Map Frame will be skipped.");
+                }
+            }
+        }
+
+        if (polygons.size() == 0) {
+            LOGGER.debug(
+                    "No GeoPDF information found on PDF during transformation.  Metacard location will not be set.");
+            return null;
+        }
+
+        StringBuilder wkt = new StringBuilder();
+        if (polygons.size() == 1) {
+            wkt.append(POLYGON);
+            wkt.append(polygons.get(0));
+            wkt.append("))");
+        } else {
+            wkt.append(MULTIPOLYGON);
+            for (int i = 0; i < polygons.size(); i++) {
+                polygons.set(i, "((" + polygons.get(i) + "))");
+            }
+            wkt.append(StringUtils.join(polygons, ","));
+            wkt.append(")");
+        }
+        return wkt.toString();
+    }
+
+    /**
+     * A PDF Neatline defines the area of a PDF image with relation to the PDF page.  If no neatline is given
+     * it is assumed that the image encompasses the entire page.  This functiong generates a NeatLine
+     * in this fashion.
+     * @param pdPage the page to generate the NeatLine
+     * @return an array of points representing the NeatLine
+     */
+    private COSArray generateNeatLineFromPDFDimensions(PDPage pdPage) {
+        COSArray neatLineArray = new COSArray();
+        neatLineArray.add(new COSString("0"));
+        neatLineArray.add(new COSString("0"));
+
+        neatLineArray.add(new COSString(pdPage.getMediaBox().getWidth() + ""));
```

Comment:
Could create two local variables for the width and height to make this a bit more readable.